crates/ironrdp-pdu/src/rdp/vc/dvc/ainput.rs

```diff
@@ -0,0 +1,259 @@
+
+    const FIXED_PART_SIZE: usize = core::mem::size_of::<u32>() * 2;
+
+    pub fn new() -> Self {
+        Self {
+            major_version: VERSION_MAJOR,
+            minor_version: VERSION_MINOR,
+        }
+    }
+}
+
+impl Default for VersionPdu {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl PduEncode for VersionPdu {
+    fn encode(&self, dst: &mut WriteCursor<'_>) -> PduResult<()> {
+        ensure_fixed_part_size!(in: dst);
+
+        dst.write_u32(self.major_version);
+        dst.write_u32(self.minor_version);
+
+        Ok(())
+    }
+
+    fn name(&self) -> &'static str {
+        Self::NAME
+    }
+
+    fn size(&self) -> usize {
+        Self::FIXED_PART_SIZE
+    }
+}
+
+impl<'de> PduDecode<'de> for VersionPdu {
+    fn decode(src: &mut ReadCursor<'de>) -> PduResult<Self> {
+        ensure_fixed_part_size!(in: src);
+
+        let major_version = src.read_u32();
+        let minor_version = src.read_u32();
+
+        Ok(Self {
+            major_version,
+            minor_version,
+        })
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, FromPrimitive, ToPrimitive)]
+pub enum ServerPduType {
+    Version = 0x01,
+}
+
+impl<'a> From<&'a ServerPdu> for ServerPduType {
+    fn from(s: &'a ServerPdu) -> Self {
+        match s {
+            ServerPdu::Version(_) => Self::Version,
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ServerPdu {
+    Version(VersionPdu),
+}
+
+impl ServerPdu {
+    const NAME: &'static str = "AInputServerPdu";
+
+    const FIXED_PART_SIZE: usize = core::mem::size_of::<u16>();
+}
+
+impl PduEncode for ServerPdu {
+    fn encode(&self, dst: &mut WriteCursor<'_>) -> PduResult<()> {
+        ensure_fixed_part_size!(in: dst);
+
+        dst.write_u16(ServerPduType::from(self).to_u16().unwrap());
+        match self {
+            ServerPdu::Version(pdu) => pdu.encode(dst),
+        }
+    }
+
+    fn name(&self) -> &'static str {
+        Self::NAME
+    }
+
+    fn size(&self) -> usize {
+        Self::FIXED_PART_SIZE
+            + match self {
+                ServerPdu::Version(pdu) => pdu.size(),
+            }
+    }
+}
+
+impl<'de> PduDecode<'de> for ServerPdu {
+    fn decode(src: &mut ReadCursor<'de>) -> PduResult<Self> {
+        ensure_fixed_part_size!(in: src);
+
+        let pdu_type = ServerPduType::from_u16(src.read_u16())
+            .ok_or_else(|| invalid_message_err!("pduType", "invalid pdu type"))?;
+
+        let server_pdu = match pdu_type {
+            ServerPduType::Version => ServerPdu::Version(VersionPdu::decode(src)?),
+        };
+
+        Ok(server_pdu)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct MousePdu {
+    pub time: u64,
+    pub flags: MouseEventFlags,
+    pub x: i32,
+    pub y: i32,
+}
+
+impl MousePdu {
+    const NAME: &'static str = "AInputMousePdu";
+
+    const FIXED_PART_SIZE: usize = core::mem::size_of::<u64>() * 2 + core::mem::size_of::<i32>() * 2;
```

Comment:
(style) Our new code style states the following about size fields:

>Use an inline comment for each field of the structure.

```rust
// GOOD
const FIXED_PART_SIZE: usize = 1 /* Version */ + 1 /* Endianness */ + 2 /* CommonHeaderLength */ + 4 /* Filler */;

// BAD
const FIXED_PART_SIZE: usize = size_of::<u8>() + size_of::<u8>() + size_of::<u16>() + size_of::<u32>();
```

Please update all `FIXED_PART_SIZE` according to the new style

Reply:
ok